packages/request-client.js/test/index.test.ts

```diff
@@ -1461,6 +1461,83 @@ describe('index', () => {
       );
       sinon.restore();
     });
+
+    it('can skip the get the balance of a request', async function(): Promise<void> {
```

Comment:
I would add a check that `getBalance` was not called. We do this to improve performance, if the `balance` field is null but `getBalance` was still called, we didn't achieve much!

Reply:
after rethinking it, We return the previous balance to avoid loosing information

Reply:
I don't understand. Can't you add a spy to `getBalance` and make sure it's not called?

Reply:
I think so too

Reply:
the payment network is nested in the class as private.

Reply:
private stuff sucks because of this :/

Reply:
Can't you solve this with any ?